frontend/src/App.js

```diff
@@ -1,15 +1,150 @@
   const [user, setUser] = useState(null);
+  const [token, setToken] = useState(null);
+
 
   useEffect(() => {
-    fetch('/api/me')
-      .then(response => response.json())
-      .then(setUser);
+
+    //event listener for getting data back from popup
+
+    window.addEventListener('message', function(event){
+
+      // checking access code available or not
+
+        if(event.data != "" || event.data !== undefined){
+      
+          // storing code 
+          let code = event.data;
+          
+          // create data variable to send in api
+
+          let data = {
+            code: code
+          };
+
+          // api for getting jwt token from backend api
+
+          fetch('/api/github/token', {
+                method: 'POST',
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(data)
+            }).then(function(response) {
+              return response.json();
+            }).then(function(data) {
+
+                // chekcing data contains token or not
+                if(data.token){
+
+                  // setting token
+                  setToken(data.token)
+
+                  // try to call api again after getting token
+                  fetch('/api/me',{ 
+                          method: "GET",
+                          headers: {
+                                    'Authorization': `token ${token}`
+                                  }
+                  })
+                  .then(response => response.json())
+                  .then(setUser)
+                  .catch(error => console.log('error============:', error));
+      
+                }
+
+            });
+          
+        }
+    }); 
+
+
+    // calling api to check token authentication wokring or not
+      fetch('/api/me')
+        .then(response => response.json())
+        .then(setUser)
+        .catch(error => console.log('error============:', error));
+
+
   }, []);
 
+
+  // // getting success data after github login
+  // const onSuccess = response => {
+  // //   console.log(response)
+  // //  axios.post('/api/github/token', {
+  // //   code:response.code
+  // //   })
+  // //   .then((response) => {
+  // //     // console token 
+  // //     console.log(response.data);
+     
+  // //   })
+  // //   .catch((error) => {
+  // //     console.error(error)
+  // //   });
+  
+  // };
+
+  // failure data in case login fails
+  // const onFailure = response => console.error(response);
+
+
+  // function for github login
+  const handleLogin = (url,id) => {
+
+    const op = PopupCenter(url, id,600,1000);
+
+  };
+
+
+ // function to open popup window at center
+
+  const PopupCenter = (url, title, w, h)  => {
+    // Fixes dual-screen position                         Most browsers      Firefox
+    var dualScreenLeft = window.screenLeft != undefined ? window.screenLeft : window.screenX;
```

Comment:
This looks like copy-pasted from somewhere. If so, can you please include the URL in a comment to where it came from? Also please use `const` instead of `var`. Thank you.